### PR TITLE
refactor(runtime/permissions): Rename permission structs

### PR DESCRIPTION
**Pull request by nayeemrmn**
- `BooleanPermission` -> `UnitPermission`: Since it's not boolean. `UnaryPermission` queries with one argument, `UnitPermission` queries with 0 and is only associated with a single `PermissionState`.
- `{Read,Write,Net}Permission` -> `{Read,Write,Net}Descriptor`: `UnaryPermission<ReadDescriptor>` doesn't stutter, more accurate.

### CODE REVIEW

File: runtime/permissions.rs [hunk 1]
```diff
@@ -81,9 +116,15 @@ pub struct UnaryPermission<T: Eq + Hash> {
...
 
 #[derive(Clone, Eq, PartialEq, Hash, Debug, Default, Deserialize)]
-pub struct ReadPermission(pub PathBuf);
+pub struct ReadDescriptor(pub PathBuf);
+
+#[derive(Clone, Eq, PartialEq, Hash, Debug, Default, Deserialize)]
+pub struct WriteDescriptor(pub PathBuf);
+
+#[derive(Clone, Eq, PartialEq, Hash, Debug, Default, Deserialize)]
+pub struct NetDescriptor(pub String, pub Option<u16>);
```
Comment (crowlKats) [inline runtime/permissions.rs:125]: why did you move the descriptor definitions all together? having them to their corresponding impl block seems better to me

Comment (nayeemrmn) [reply runtime/permissions.rs:125]: I had trouble finding them initially, and this was where I looked for them to be so it felt right. Probably because the impl blocks are for `UnaryPermission<_>`.

Comment (crowlKats) [reply runtime/permissions.rs:125]: hm, alright. there is also some impls for the NetDescriptor directly, so maybe you wanna move those up as well

Comment (crowlKats) [reply runtime/permissions.rs:125]: (you forgot the display impl for net descriptor)

**Review by crowlKats — APPROVED** [review]
LGTM

**Review by ry — APPROVED** [review]
LGTM
